Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -1022,11 +1022,16 @@ private bool ExecuteOnAllServicesInOrder(Action<IMixedRealityService> execute)
                     return false;
                 }
 
-                var services = MixedRealityServiceRegistry.GetAllServices();
-                int length = services.Count;
-                for (int i = 0; i < length; i++)
+                try
                 {
-                    execute(services[i]);
+                    foreach (IMixedRealityService service in MixedRealityServiceRegistry.GetAllServices())
```

Comment:
Same thing here for foreach -> for (I believe you're already going back to for). The memory alloc is going to be painful over each frame